cmd/clusterctl/client/cluster/components.go

```diff
@@ -168,19 +169,30 @@ func (p *providerComponents) Delete(options DeleteOptions) error {
 			namespacesToDelete.Insert(obj.GetName())
 		}
 
+		// If the resource is part of the inventory for clusterctl don't delete it at this point as losing this information makes the
+		// upgrade function non-reentrant. Instead keep the inventory objects around until the upgrade is finished and working and
+		// delete them at the end of the upgrade flow.
+		isInventory := obj.GroupVersionKind().GroupKind().String() == providerGroupKind
```

Comment:
You have to check for group and kind only, given that version might change (we are not yet graduated).
Also group is defined in the clusterctl API, so you can reuse it.

Reply:
I think this is only checking the group and kind though right? I'm accessing groupKind through the GroupVersionKind, but that's only for access.

Reply:
you are right... 😅